### ISSUE TITLE
`rust-analyzer` can't parse leading `|` in patterns

### ISSUE THREAD

**Issue by jinxdash** [rust-lang/rust-analyzer#13094]
**rust-analyzer version**: 0.3.1178-standalone (latest)

**rustc version**: 1.60.0

```rs
#[derive(Debug)]
enum KeyCode {
    Up,
    Down,
	Left,
}

fn main() {
    foo(KeyCode::Up);
}

fn foo(x: KeyCode) {
	if let (a @ (  KeyCode::Up | KeyCode::Down),) = (x,) {
		println!("{:?}", a);
	}
}

fn bar(x: KeyCode) {
	if let (a @ (| KeyCode::Up | KeyCode::Down),) = (x,) {
                  // ^ Syntax Error: expected a pattern rust-analyzer
		println!("{:?}", a);
	}
}
```

Actual: Syntax Error at leading `|` in fn bar

Expected: No Syntax Error (see [rust spec](https://doc.rust-lang.org/reference/patterns.html))

EDIT: looks like this is a duplicate of #12894
